llvm/lib/CodeGen/AsmPrinter/DwarfFile.cpp

```diff
@@ -94,7 +122,44 @@ unsigned DwarfFile::computeSizeAndOffset(DIE &Die, unsigned Offset) {
                                       Offset);
 }
 
-void DwarfFile::emitAbbrevs(MCSection *Section) { Abbrevs.Emit(Asm, Section); }
+void DwarfFile::emitAbbrevs(MCSection *Section) {
+  // If the abbreviation vector only has one .debug_abbrev contribution and the
+  // contribution has non-zero abbreviations in use, emit it without emitting
+  // any labels. This is done because there are some targets do not support
+  // labels inside debug sections.
+  if (Abbrevs.size() == 1 && !Abbrevs[0]->isAbbreviationListEmpty()) {
+    Asm->OutStreamer->SwitchSection(Section);
+    Abbrevs[0]->Emit(Asm);
+    return;
+  }
+  for (unsigned I = 0; I < Abbrevs.size(); I++) {
+    // If the abbreviation contribution is empty, do not
+    // switch sections or emit labels.
+    if (Abbrevs[I]->isAbbreviationListEmpty())
+      continue;
+
+    Asm->OutStreamer->SwitchSection(Section);
+    MCSymbol *AbbrevStartSym = getOrCreateAbbrevSectionStart();
+    // Do not emit the same label again, AbbrevStartSym should be the same for
+    // all abbreviation contributions, it denotes the start of the
+    // abbreviation Section.
+    if (!AbbrevStartEmitted) {
+      AbbrevStartEmitted = true;
+      Asm->OutStreamer->emitLabel(AbbrevStartSym);
+    }
+      if (CUs.size()) {
```

Comment:
clang-format

Reply:
When would CUs.size() be false?

Reply:
This is no longer needed, it was a temporary check added because there are some tests where the CUs are empty and so are the abbrevs, but the way the code was written, it would have still tried to do `CUs[i]` which would lead to crashes, it has been removed. Good catch!